pkg/pipeline/extract/extract_aggregate_test.go

```diff
@@ -0,0 +1,172 @@
+			"count":           "2",
+		},
+		{
+			"name":                  "bandwidth_count",
+			"record_key":            "",
+			"by":                    "service",
+			"aggregate":             "http",
+			"service":               "http",
+			"operation":             aggregate.Operation(aggregate.OperationCount),
+			"value":                 "2.000000",
+			"bandwidth_count_value": "2.000000",
+			"recentRawValues":       []float64{1.0, 1.0},
+			"count":                 "2",
+		},
+		{
+			"name":                  "bandwidth_count",
+			"record_key":            "",
+			"by":                    "service",
+			"aggregate":             "tcp",
+			"service":               "tcp",
+			"operation":             aggregate.Operation(aggregate.OperationCount),
+			"value":                 "2.000000",
+			"bandwidth_count_value": "2.000000",
+			"recentRawValues":       []float64{1.0, 1.0},
+			"count":                 "2",
+		},
+	}
+	actualAggs1 := extractAggregate.Extract(input1)
+	require.ElementsMatch(t, expectedAggs1, actualAggs1)
+
+	input2 := []config.GenericMap{
+		{"service": "http", "bytes": 30.0},
+		{"service": "tcp", "bytes": 4.0},
+		{"service": "tcp", "bytes": 5.0},
+	}
+	expectedAggs2 := []config.GenericMap{
+		{
+			"name":            "bandwidth",
+			"record_key":      "bytes",
+			"by":              "service",
+			"aggregate":       "http",
+			"service":         "http",
+			"operation":       aggregate.Operation(aggregate.OperationSum),
+			"value":           "60.000000",
+			"bandwidth_value": "60.000000",
+			"recentRawValues": []float64{30.0},
+			"count":           "3",
+		},
+		{
+			"name":            "bandwidth",
+			"record_key":      "bytes",
+			"by":              "service",
+			"aggregate":       "tcp",
+			"service":         "tcp",
+			"operation":       aggregate.Operation(aggregate.OperationSum),
+			"value":           "12.000000",
+			"bandwidth_value": "12.000000",
+			"recentRawValues": []float64{4.0, 5.0},
+			"count":           "4",
+		},
+		{
+			"name":                  "bandwidth_count",
+			"record_key":            "",
+			"by":                    "service",
+			"aggregate":             "http",
+			"service":               "http",
+			"operation":             aggregate.Operation(aggregate.OperationCount),
+			"value":                 "3.000000",
+			"bandwidth_count_value": "3.000000",
+			"recentRawValues":       []float64{1.0},
+			"count":                 "3",
+		},
+		{
+			"name":                  "bandwidth_count",
+			"record_key":            "",
+			"by":                    "service",
+			"aggregate":             "tcp",
+			"service":               "tcp",
+			"operation":             aggregate.Operation(aggregate.OperationCount),
+			"value":                 "4.000000",
+			"bandwidth_count_value": "4.000000",
+			"recentRawValues":       []float64{1.0, 1.0},
+			"count":                 "4",
+		},
+	}
+	actualAggs2 := extractAggregate.Extract(input2)
+	require.ElementsMatch(t, expectedAggs2, actualAggs2)
+
+}
```

Comment:
@ronensc general remark, maybe put the use cases in some for loop and just run them all of them against `extractAggregate` --- this is very common for tests --- look here for an example https://github.com/netobserv/flowlogs-pipeline/commit/b24be79c22c1b982f4dbd25ea3e2e777f22ae0fc#diff-168486be9e126a8de1f0588534a9bf2fd49261555cbdc3759a5be2b1d2825cd3R26

Reply:
@eranra The test tests `extract_aggregate` as a whole. It can be thought of as an integration test between `extract_aggregate.go` and `aggregate.go` and `aggregates.go`.
The test sends flows in 2 batches and verifies the extractor's output after each batch. The output of the 2nd batch depends on the 1st batch. I'll check if the referred pattern is suitable here as well.